src/Event/OgAdminRoutesEvent.php

```diff
@@ -40,8 +41,9 @@ public function getRoutes($entity_type_id) {
 
       $routes_info[$name] = $route_info;
 
-      // Add default values.
-      $routes_info[$name] += [
+      // Add default values. NestedArray::mergeDeep allows deep data to not be
+      // overwritten with the defaults.
+      $routes_info[$name] = NestedArray::mergeDeep($routes_info[$name], [
```

Comment:
This actually overwrites important parameters such as `_og_user_access_group`.
See screenshot below. Custom route defined as:

```
    $routes_info['broadcast'] = [
      'controller' => '\Drupal\og_broadcast\Controller\BroadcastController::groupForm',
      'title' => 'Broadcast',
      'description' => 'Send broadcast message to members',
      'path' => 'broadcast',
      'requirements' => [
        '_og_user_access_group' => 'send broadcast message',
      ],
    ];
```

After `mergeDeep`, `_og_user_access_group` is "administer group".

![screen_shot_2017-11-03_at_14 47 43](https://user-images.githubusercontent.com/31622/32379795-6b6a3144-c0a6-11e7-90f1-f2219624f8cb.jpg)

Reply:
swapping the arrays around fixes the issue. Here's a diff https://gist.github.com/zerolab/fa9cf684cdf0afe616bc7f23629f1603

Reply:
You must be using an older version of the PR. The current code is like this:

```  public function getRoutes($entity_type_id) {
    $routes_info = [];

    foreach ($this->routesInfo as $name => $route_info) {

      $routes_info[$name] = $route_info;

      // Add default values.
      $routes_info[$name] += [
        'description' => '',

        'requirements' => [
          '_og_user_access_group' => 'administer group',
        ],

        'options' => [
          'parameters' => [
            $entity_type_id => ['type' => 'entity:' . $entity_type_id],
          ],
          // The above parameters doesn't send the entity,
          // so we will have to use the Route matcher to extract it.
          '_og_entity_type_id' => $entity_type_id,
          '_admin_route' => TRUE,
        ],

        // Move the title and controller under the "defaults" key.
        'defaults' => [
          '_controller' => $route_info['controller'],
          '_title' => $route_info['title'],
        ],
      ];
    }

    return $routes_info;
  }
```

which I think results in the same as your patch.

Reply:
Thanks! Added this in.

Reply:
👍